src/com/shmuelzon/HomeAssistantFloorPlan/Controller.java

```diff
@@ -962,14 +962,30 @@ private Point2d getCenterOfStateIcons(Set<Entity> entities) {
     private void separateStateIcons(Set<Entity> entities) {
         final double STEP_SIZE = 2.0;
 
-        Point2d centerPostition = getCenterOfStateIcons(entities);
+        if (entities.isEmpty()) {
+            return;
+        }
+
+        Entity entityToMove = entities.iterator().next();
+        // Round position-values because the entity's stored position can differ fractionally from the position in SH3D.
+        entityToMove.position.set(Math.round(entityToMove.position.x), Math.round(entityToMove.position.y));
```

Comment:
Didn't you round this in another commit? Why do we need it twice (and at all)?

Reply:
Actually, I think I did because the same set of entities is passed to the different methods.
But from the method's point of view, the method doesn't know about it.

Perhaps it's better to extract another method for rounding the values, if rounding is required. See https://github.com/shmuelzon/home-assistant-floor-plan/pull/101#discussion_r1898518024